pkg/types/azure/validation/machinepool.go

```diff
@@ -12,5 +12,14 @@ func ValidateMachinePool(p *azure.MachinePool, fldPath *field.Path) field.ErrorL
 	if p.OSDisk.DiskSizeGB < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskSizeGB"), p.OSDisk.DiskSizeGB, "Storage DiskSizeGB must be positive"))
 	}
+
+	if p.OSDisk.DiskType != "" {
+		//"UltraSSD_LRS": "",
+		_, found := map[string]string{"Standard_LRS": "", "StandardSSD_LRS": "", "Premium_LRS": ""}[p.OSDisk.DiskType]
+		if !found {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("diskType"), p.OSDisk.DiskType, "Storage DiskType value invalid"))
```

Comment:
include the list of supported values in the message.

Also i think there is https://www.godoc.org/k8s.io/apimachinery/pkg/util/validation/field#NotSupported